src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1824,8 +1824,12 @@ static void updateAttrInNotifyCer
         if (caP->compoundValueP != NULL)
         {
           delete caP->compoundValueP;
+          caP->compoundValueP = NULL;
         }
-        caP->compoundValueP = targetAttr->compoundValueP == NULL ? NULL : targetAttr->compoundValueP->clone();
+
+        // Steal compound value from targetAttr
+        caP->compoundValueP        = targetAttr->compoundValueP;
```

Comment:
I understand that changing strategy from cloning to stealling may involve some "move" in the release logic. Has you check that?